x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRankBuilder.java

```diff
@@ -103,6 +106,62 @@ public QueryPhaseRankCoordinatorContext buildQueryPhaseCoordinatorContext(int si
         return new RRFQueryPhaseRankCoordinatorContext(size, from, rankWindowSize(), rankConstant);
     }
 
+    @Override
+    public Explanation explainHit(Explanation baseExplanation, RankDoc rankDoc, List<String> queryNames) {
```

Comment:
This is so frustrating. The parser only support for retrievers is really hampering us from building this further into anything logical.

Really, all the explanation stuff SHOULD live in retrievers. Continuing to build on this bad interface is frustrating. There is no knowledge of relationship, everything is flattened out.

Reply:
+1 on this. I would too opt to have this on the `retriever` level, but unfortunately that's not an option.. And this gets even more frustrating when you want to build nested things, as the concept of "retriever" disappears right after parsing.. So any nested sub-retrievers, working on different outputs etc, is a bit of a pain to work with :(

Reply:
@pmpailis trying to figure out how to do explain for any nested thing will be rough. The only way to do this is if when we flatten it out, the original order is respected, then it turns into a iteration problem, not a recursive one.